homeassistant/components/otbr/config_flow.py

```diff
@@ -23,22 +25,36 @@ async def async_step_user(
         if self._async_current_entries():
             return self.async_abort(reason="single_instance_allowed")
 
+        errors = {}
+
         if user_input is not None:
-            return self.async_create_entry(
-                title="Thread",
-                data={"url": user_input[CONF_URL]},
-            )
+            url = user_input[CONF_URL]
+            api = python_otbr_api.OTBR(url, async_get_clientsession(self.hass), 10)
+            try:
+                await api.get_active_dataset_tlvs()
+            except python_otbr_api.OTBRError:
+                errors["base"] = "cannot_connect"
+            else:
+                await self.async_set_unique_id(DOMAIN)
+                return self.async_create_entry(
+                    title="Thread",
+                    data={"url": url},
```

Comment:
Already guarded by the schema, so we can directly use it.

```suggestion
                    data=user_input,
```